docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+    ``list_prefix("meta/root/")``. All keys represent either explicit group or
+    arrays. All intermediate prefixes ending in a ``/`` are implicit
+    groups.
+
+**Erase a group or array**
+
+    To erase an array at path `P`:
+      - erase the metadata document for the array, ``erase(array_meta_key(P))``
+      - erase all data keys which prefix have path pointing to this array,
+        ``erase_prefix("data/root" + P + "/")``
+
+    To erase an implicit group at path `P`:
+      - erase all nodes under this group - it should be sufficient to
+        perform ``erase_prefix("meta/root" + P + "/")`` and
+        ``erase_prefix("data/root" + P + "/")``.
+
+    To erase an explicit group at path `P`:
+      - erase the metadata document for the group, ``erase(group_meta_key(P))``
+      - erase all nodes under this group - it should be sufficient to
+        perform ``erase_prefix("meta/root" + P + "/")`` and
+        ``erase_prefix("data/root" + P + "/")``.
+
+**Determine if a node exists**
+
+    To determine if a node exists at path ``P``, try in the following
+    order ``get(array_meta_key(P))`` (success implies an array at
+    ``P``); ``get(group_meta_key(P))`` (success implies an explicit
+    group at ``P``); ``list_dir("meta/root" + P + "/")`` (non-empty
+    result set implies an implicit group at ``P``).
+
+    .. note::
+       For listable store, ``list_dir(parent(P))`` can be an alternative.
+
+
+Storage transformers
+====================
+
+A Zarr storage transformer allows to change the zarr-compatible data before storing it.
+The stored transformed data is restored to its original state whenever data is requested
+by the Array. Storage transformers can be configured per array via the
+`storage_transformers`_ name in the `array metadata`_. Storage transformers which do
+not change the storage layout (e.g. for caching) may be specified at runtime without
+adding them to the array metadata.
+
+A storage transformer serves the same `abstract store interface`_ as the store_.
+However, it should not persistently store any information necessary to restore the original data,
+but instead propagates this to the next storage transformer or the final store.
+From the perspective of an array or a previous stage transformer both store and storage transformer follow the same
+protocol and can be interchanged regarding the protocol. The behaviour can still be different,
+e.g. requests may be cached or the form of the underlying data can change.
+
+Storage transformers may be stacked to combine different functionalities:
+
+.. mermaid::
+
+    graph LR
+      Array --> t1
+      subgraph stack [Storage transformers]
+        t1[Transformer 1] --> t2[...] --> t3[Transformer N]
+      end
+      t3 --> Store
+
+A fixed set of storage providers is recommended for implementation with this specification:
+
+
+Predefined storage transformers
+-------------------------------
+
+- :ref:`sharding-storage-transformer-v1`
```

Comment:
@alimanfoo So far no extensions are part of the ZEP0001, right? Should we remove such references then? We could instead leave them in and mention that the current set of extensions is still pending and needs to go through the ZEP process. I'd tend towards the latter, to keep concrete examples of the extensions in the spec docs.